### PR TITLE
Upgrading to jdk-21+26.

### PR DESCRIPTION
**Pull request by dbalek**
Upgrading to jdk-21+26.

### CODE REVIEW

File: make/langtools/netbeans/nb-javac/src/META-INF/upgrade/nbjavac.hint [hunk 1]
```diff
@@ -337,3 +355,36 @@ sun.reflect.annotation.AnnotationTypeMismatchExceptionProxy
 nbjavac.sun.annotation.AnnotationTypeMismatchExceptionProxy
 ;;
 
+$cs.isEmpty() :: $cs instanceof java.lang.CharSequence
+=>
+($cs.length() == 0)
+;;
+
+$coll.toArray($type[]::new) :: inClass("com.sun.tools.javac.comp.Flow")
```
Comment (jtulach) [inline make/langtools/netbeans/nb-javac/src/META-INF/upgrade/nbjavac.hint:363]: This is probably needed in order to run on JDK-8, right? The [toArray(IntFunction)](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/Collection.html#toArray(java.util.function.IntFunction)) has only been added in JDK-11.


Why there is a restriction `inClass("com.sun.tools.javac.comp.Flow")` - this seems like a legitimate generic rule to use.

**Review by jtulach — APPROVED** [review]
Great work, Dušane!

I can see that the [NetBeans compilation is failing](https://github.com/JaroslavTulach/nb-javac/actions/runs/5398177630/jobs/9803680441?pr=18#step:10:12158) on a new warning. We need a fix on the NetBeans side (probably adding a `@SuppressWarnings` annotation) and then we need to change the [commit hash we test against](https://github.com/JaroslavTulach/nb-javac/blob/master/.github/workflows/netbeans.yml#L33).

Once fixed and both gates are green, feel free to integrate.

File: make/langtools/netbeans/nb-javac/src/META-INF/upgrade/nbjavac.hint (hunk 1 above)
Comment (dbalek) [reply make/langtools/netbeans/nb-javac/src/META-INF/upgrade/nbjavac.hint:363]: An attempt to make the rule generic resulted in build failures.

File: .github/workflows/netbeans.yml [hunk 2]
```diff
@@ -30,12 +37,12 @@ jobs:
         run: |
             git clone https://github.com/apache/netbeans
             cd netbeans
-            git checkout b03de5696eb2c18f13877dabbf8641d92274dc3a
+            git checkout 0a84604d51f9e8b560c060b8e26cdb3ee3141804
       - name: Build NetBeans with latest nb-javac
-        run: JAVA_HOME=`cat jdk8` ant -f netbeans build -Dnbjavac.class.path=`pwd`/make/langtools/netbeans/nb-javac/dist/nb-javac-*.jar
```
Comment (jtulach) [inline .github/workflows/netbeans.yml:35]: De gustibus non est disputandum: https://github.com/apache/netbeans/pull/6154/files#diff-7829468e86c1cc5d5133195b5cb48e1ff6c75e3e9203777f6b2e379d9e4882b3L165

**Comment by jtulach** [comment]
Bits are [staged at oss sonatype](https://oss.sonatype.org/content/repositories/comdukescript-1161). Please confirm they are OK.

**Comment by jtulach** [comment]
Bits of `nb-javac` [jdk-21+26](https://repo1.maven.org/maven2/com/dukescript/nbjavac/nb-javac/jdk-21+26/) are available at [maven central](https://repo1.maven.org/maven2/com/dukescript/nbjavac/nb-javac/jdk-21+26/).

**Pull request merged**